src/main/java/com/conveyal/datatools/manager/controllers/api/DeploymentController.java

```diff
@@ -494,22 +492,29 @@ private static String deploy (Request req, Response res) {
+        JsonManager<Deployment> slimJson = new JsonManager<>(Deployment.class, JsonViews.UserInterface.class);
+        JsonManager<Deployment> fullJson = new JsonManager<>(Deployment.class, JsonViews.UserInterface.class);
+        fullJson.addMixin(Deployment.class, Deployment.DeploymentWithEc2InstancesMixin.class);
+
+        post(apiPrefix + "secure/deployments/:id/deploy/:target", DeploymentController::deploy, slimJson::write);
         post(apiPrefix + "secure/deployments/:id/deploy/", ((request, response) -> {
             logMessageAndHalt(request, 400, "Must provide valid deployment target name");
             return null;
-        }), json::write);
+        }), slimJson::write);
         options(apiPrefix + "secure/deployments", (q, s) -> "");
         get(apiPrefix + "secure/deployments/:id/download", DeploymentController::downloadDeployment);
         get(apiPrefix + "secure/deployments/:id/artifact", DeploymentController::downloadBuildArtifact);
-        get(apiPrefix + "secure/deployments/:id/ec2", DeploymentController::fetchEC2InstanceSummaries, json::write);
-        delete(apiPrefix + "secure/deployments/:id/ec2", DeploymentController::terminateEC2InstanceForDeployment, json::write);
-        get(apiPrefix + "secure/deployments/:id", DeploymentController::getDeployment, json::write);
-        delete(apiPrefix + "secure/deployments/:id", DeploymentController::deleteDeployment, json::write);
-        get(apiPrefix + "secure/deployments", DeploymentController::getAllDeployments, json::write);
-        post(apiPrefix + "secure/deployments", DeploymentController::createDeployment, json::write);
-//        post(apiPrefix + "secure/deployments/:id/ec2", DeploymentController::addEC2InstanceToDeployment, json::write);
-        put(apiPrefix + "secure/deployments/:id", DeploymentController::updateDeployment, json::write);
-        post(apiPrefix + "secure/deployments/fromfeedsource/:id", DeploymentController::createDeploymentFromFeedSource, json::write);
+        get(apiPrefix + "secure/deployments/:id/ec2", DeploymentController::fetchEC2InstanceSummaries, slimJson::write);
+        delete(apiPrefix + "secure/deployments/:id/ec2", DeploymentController::terminateEC2InstanceForDeployment, slimJson::write);
+        get(apiPrefix + "secure/deployments/:id", DeploymentController::getDeployment, fullJson::write);
+        delete(apiPrefix + "secure/deployments/:id", DeploymentController::deleteDeployment, fullJson::write);
+        get(apiPrefix + "secure/deployments", DeploymentController::getAllDeployments, slimJson::write);
+        post(apiPrefix + "secure/deployments", DeploymentController::createDeployment, fullJson::write);
+        put(apiPrefix + "secure/deployments/:id", DeploymentController::updateDeployment, fullJson::write);
+        post(apiPrefix + "secure/deployments/fromfeedsource/:id", DeploymentController::createDeploymentFromFeedSource, fullJson::write);
```

Comment:
It is nearly impossible to tell what this PR is about anymore since it contains a whole bunch of stuff from the dev branch.